Sources/SmileID/Classes/Camera/CameraViewController.swift

```diff
@@ -39,3 +55,24 @@ extension PreviewView: FaceDetectorDelegate {
         return previewLayer.layerRectConverted(fromMetadataOutputRect: rect)
     }
 }
+
+extension PreviewView: RectangleDetectionDelegate {
+    func didDetectQuad(quad: Quadrilateral?, _ imageSize: CGSize) {
+        guard let quad else {
+            quadView.removeQuadrilateral()
+            return
+        }
+        let portraitImageSize = CGSize(width: imageSize.height, height: imageSize.width)
+        let scaleTransform = CGAffineTransform.scaleTransform(forSize: portraitImageSize,
+                                                              aspectFillInSize: quadView.bounds.size)
+        let scaledImageSize = imageSize.applying(scaleTransform)
+        let rotationTransform = CGAffineTransform(rotationAngle: CGFloat.pi / 2.0)
+        let imageBounds = CGRect(origin: .zero, size: scaledImageSize).applying(rotationTransform)
+        let translationTransform = CGAffineTransform.translateTransform(fromCenterOfRect: imageBounds,
+                                                                        toCenterOfRect: quadView.bounds)
+
+        let transforms = [scaleTransform, rotationTransform, translationTransform]
+        let transformedQuad = quad.applyTransforms(transforms)
+        quadView.drawQuadrilateral(quad: transformedQuad, animated: true)
+    }
+}
```

Comment:
We instead need logic to detect whether the quadrilateral is centered in frame, whether it is skewed or not, and to update the overlay if the document is properly positioned

Reply:
Bump